go/vt/vtgate/schema/tracker.go

```diff
@@ -134,21 +134,18 @@ func (t *Tracker) loadViews(conn queryservice.QueryService, target *querypb.Targ
 		return nil
 	}
 
-	fvRes, err := conn.GetSchema(t.ctx, target, querypb.SchemaTableType_VIEWS, nil)
-	if err != nil {
-		return err
-	}
-
 	t.mu.Lock()
 	defer t.mu.Unlock()
 	// We must clear out any previous view definition before loading it here as this is called
-	// whenever a shard's primary tablet starts and sends the initial signal. Without
-	// clearing out the previous view definition removes any dropped views.
+	// whenever a shard's primary tablet starts and sends the initial signal.
+	// This is needed clear out any stale view definitions.
 	t.clearKeyspaceViews(target.Keyspace)
-	t.updateViews(target.Keyspace, fvRes)
-	log.Infof("finished loading views for keyspace %s. Found %d views", target.Keyspace, len(fvRes))
 
-	return nil
+	return conn.GetSchema(t.ctx, target, querypb.SchemaTableType_VIEWS, nil, func(schemaRes *querypb.GetSchemaResponse) error {
+		t.updateViews(target.Keyspace, schemaRes.TableDefinition)
+		log.Infof("finished loading views for keyspace %s. Found %d views", target.Keyspace, len(schemaRes.TableDefinition))
```

Comment:
This log line is now incorrect. You can have the view definitions coming in separate packets, so you don't want to print 2 lines saying it finished loading the views for a keyspace and reporting different number of views each time, while the correct count is the sum of both of them

Reply:
I thought about that. moving it out.